### ISSUE TITLE
404 from pythonhosted

### ISSUE THREAD

**Issue by charmoniumQ** [nix-community/poetry2nix#310]
When `poetry2nix` tries to install [MarkupSafe](https://pypi.org/project/MarkupSafe/), the curl from files.pythonhosted.org fails with a 404. Perhaps it is getting the wrong URL. This is the same error as #203 , but in this case, the library is actively developed, and it seems to work with Poetry.

# My debugging

I think this is because [`predictURLFromPypi`](https://github.com/nix-community/poetry2nix/blob/a5d82ecb020e6c058c0a515520a4e22133278b92/lib.nix#L90) is wrong. `predictURLFromPypi` will return `https://files.pythonhosted.org/packages/source/m/MarkupSafe/MarkupSafe-2.0.1.tar.gz`, but somehow the file that MarkupSafe hosts is at the URL `https://files.pythonhosted.org/packages/bf/10/ff66fea6d1788c458663a84d88787bae15d45daa16f6b3ef33322a51fc7e/MarkupSafe-2.0.1.tar.gz` (PyPI lists all candidates [here](https://pypi.org/project/MarkupSafe/#files)). I could be totally off here; I'm just guessing.

I don't know enough about PyPI infrastructure to know how one should ask for this URL.

# Minimal (non) Working Example

1. Start with this `pyproject.toml`:

```
[tool.poetry]
name = "test"
version = "0.1.0"
description = ""
authors = ["Samuel Grayson <sam@example.com>"]

[tool.poetry.dependencies]
python = "^3.8"
MarkupSafe = "^2.0.1"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```
2. Run `poetry update`. All good.
3. Run `poetry shell` and `python -c 'import markupsafe; print(markupsafe.escape("<hello>"))'`. See `&lt;hello&gt;`. All good.
4. Write this `shell.nix`:

```
{ pkgs ? import <nixpkgs> {}}:
pkgs.poetry2nix.mkPoetryEnv {
    projectDir = ./.;
}
```

5. Run `nix-shell shell.nix`. See:

```
trying https://files.pythonhosted.org/packages/source/m/MarkupSafe/MarkupSafe-2.0.1.tar.gz
curl: (22) The requested URL returned error: 404
error: cannot download MarkupSafe-2.0.1.tar.gz from any mirror
```

**Comment by ryneeverett** [comment]
This is resolved by #304.

**Comment by adisbladis** [comment]
#304 is merged so this should now be resolved.
